Raven.Database/Bundles/Replication/Controllers/ReplicationController.cs

```diff
@@ -562,7 +578,14 @@ private void SaveReplicationSource(string src, string lastEtag, int batchSize, s
                                  new RavenJObject(), null);
                 });
             }
-
+            //We don't want to delay large batches of attachments otherwise we will endup with alot of data that needs to get replicated.
+            if (Request.Content.Headers.ContentLength > Database.Configuration.Replication.ReplicationPropagationDelaySizeInBytes)
+            {
```

Comment:
but size is unknown because the documents are not fetched from the disk they are written directly to the prefetcher queue.